pkg/render/intrusiondetection/dpi/dpi.go

```diff
@@ -0,0 +1,359 @@
+// Copyright (c) 2021 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package dpi
+
+import (
+	"fmt"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/common"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/ptr"
+	"github.com/tigera/operator/pkg/render"
+	"github.com/tigera/operator/pkg/render/common/configmap"
+	"github.com/tigera/operator/pkg/render/common/meta"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	"github.com/tigera/operator/pkg/render/common/secret"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	DeepPacketInspectionNamespace = "tigera-dpi"
+	DeepPacketInspectionName      = "tigera-dpi"
+	DefaultMemoryLimit            = "1Gi"
+	DefaultMemoryRequest          = "100Mi"
+	DefaultCPULimit               = "1"
+	DefaultCPURequest             = "100m"
+)
+
+type DPIConfig struct {
+	IntrusionDetection *operatorv1.IntrusionDetection
+	Installation       *operatorv1.InstallationSpec
+	NodeTLSSecret      *corev1.Secret
+	TyphaTLSSecret     *corev1.Secret
+	TyphaCAConfigMap   *corev1.ConfigMap
+	PullSecrets        []*corev1.Secret
+	Openshift          bool
+	HasNoLicense       bool
+}
+
+func DPI(cfg *DPIConfig) render.Component {
+	return &dpiComponent{cfg: cfg}
+}
+
+type dpiComponent struct {
+	// Given configuration.
```

Comment:
Nit, I love comments, but only when they add some kind of value. I think the fact that this has "Config" in the name that this is some sort of "given configuration". If you wanted to describe this as "the configuration passed in by the caller", I think that would add some value.